src/Dialect/ONNX/ONNXOps/Math/DFT.cpp

```diff
@@ -24,34 +25,43 @@ using namespace onnx_mlir;
 
 namespace onnx_mlir {
 
-template <>
-LogicalResult ONNXDFTOpShapeHelper::computeShape() {
-  ONNXDFTOpAdaptor operandAdaptor(operands, op->getAttrDictionary());
+template <typename OP_TYPE>
+LogicalResult ONNXGenericDFTOpShapeHelper<OP_TYPE>::customComputeShape(
+    IndexExpr &axis) {
+  typename OP_TYPE::Adaptor operandAdaptor(operands, op->getAttrDictionary());
+
   // Get info about input data operand.
   Value input = operandAdaptor.getInput();
   // Get the rank to compensate for N dimensions.
   int64_t rank = createIE->getShapedTypeRank(input);
 
-  // Axis is a required attribute and should have a default value of 1.
-  int64_t axis = operandAdaptor.getAxis();
+  // Check if the dimension for axis is a literal and in range.
+  if (!axis.isLiteral())
+    return op->emitError("Can not perform Discrete Fourier Transform on "
+                         "dynamic dimensions at this time");
```

Comment:
Add a "ToFix" comment and connect it with the issue you opened.

Reply:
Okay got it!